### ISSUE TITLE
Bug report: Auto-accessibility with JS kits based on webfonts doesn't work with full-style class names

### ISSUE THREAD

**Issue by rmgodfrey** [FortAwesome/Font-Awesome#19448]
When I try copying the examples from [the documentation](https://fontawesome.com/docs/web/dig-deeper/accessibility#auto-accessibility-with-kits), auto-accessibility doesn't work. But when I replace `fa-solid` with `fas`, it works as advertised. Note that my kit is set to version 6.x, so it's not simply a matter of using a version 5 kit.

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community and thanks for this report.

I can confirm and I was able to replicate: https://jsfiddle.net/tagliala/3otv4hs5/3/

It only happens with CSS implementation

Let's assign @robmadole

**Comment by stevelacey** [comment]
@tagliala I am seeing the bug in the JS implementation, which is what your fiddle is showing, is it not?

There is no way to fix this while using the cdn, so I've imported the js into my project, and replaced this array: `["fa","fab","fas","far","fal","fad","fak"]` with `["fa-brands","fa-solid","fa-regular","fa-light","fa-duotone","fa-kit","fa-thin","fa","fab","fas","far","fal","fad","fak"]` which seems to fix the problem.

**Comment by rmgodfrey** [comment]
I think by "CSS implementation" @tagliala might mean the web font implementation, as opposed to the SVG one.

**Comment by tagliala** [comment]
>  which is what your fiddle is showing, is it not?

I'm using a Kit based on CSS, if you are experiencing this issue with the JS implementation, please fill out our [bug report template for web applications](https://github.com/FortAwesome/Font-Awesome/issues/new?assignees=&labels=bug%2Cneeds-triage&template=100_web_bug_report.yml&title=Bug%3A+)

**Comment by stevelacey** [comment]
@tagliala your fiddle references the js kit, not the css one, and this bug report already describes the bug correctly, the css kit is not meant to support auto-accessibility, only the js one is, and that is what is broken

**Comment by tagliala** [comment]
Hi @stevelacey

Sorry for the confusion

> the css kit is not meant to support auto-accessibility

Correct

> only the js one is

I'm using a JS kit using webfont implementation (JS loads the CSS)

<img width="430" alt="image" src="https://user-images.githubusercontent.com/556268/229141791-e018a307-6116-4f4a-b4e5-79e811a081f7.png">

I remember that I wasn't able to replicate with a JS kit using the JS implementation

**Comment by rmgodfrey** [comment]
Can confirm that the SVG implementation of auto-accessibility works fine. I should mention that the documentation has changed a little bit since I filed this bug, which might be adding to the confusion somewhat. (There was no such thing as a "CSS-only kit" at the time I filed this bug, or at least if there was such a thing it went by a different name. So it was clearer back then that "CSS implementation" referred to the webfont implementation.)
